File: python/Card.py
# Card.py

class Card:

    # System Ranking  Diamonds -> Clubs -> Hearts -> Spades. low -> high
    SUITS = "DCHS"

    # constructor
    def __init__(self, rank: int, suit: str):
        self.rank = rank
        self.suit = suit    # one character string

    # getters
    def get_rank(self) -> int:  # "->" is a return type hint for user to know
        return self.rank

    def get_suit(self) -> str:
        return self.suit

    # convert to the poker rank. Numbers into letters - for random gamemode
    def __str__(self) -> str:
        if self.rank == 14:
            rank_str = "A"
        elif self.rank == 13:
            rank_str = "K"
        elif self.rank == 12:
            rank_str = "Q"
        elif self.rank == 11:
            rank_str = "J"
        else:
            rank_str = str(self.rank)
        return f"{rank_str}{self.suit}"

    # create a card object from a string card "10D" - for file gamemode
    @staticmethod
    def parse(card_str: str) -> "Card":
        suit = card_str[-1].upper()
        rank_part = card_str[:-1]
        if rank_part == "A":
            rank = 14
        elif rank_part == "K":
            rank = 13
        elif rank_part == "Q":
            rank = 12
        elif rank_part == "J":
            rank = 11
        else:
            rank = int(rank_part)
        return Card(rank, suit)

File: C#/Card.cs
// Card.cs
using System;

public class Card {
	
	private int rank;
	private char suit;

	private const string SUITS = "DCHS";

	// constructor
	public Card(int rank, char suit) {
		this.rank = rank;
		this.suit = suit;
	}

	// getters
	public int GetRank() {
		return rank;
	}
		
	public char GetSuit() {
		return suit;	
	}
	
	// convert numbers into letters for poker.
	public override string ToString() {
		string rankStr;
		switch (rank) {
			case 14: rankStr = "A"; break;
			case 13: rankStr = "K"; break;
			case 12: rankStr = "Q"; break;
			case 11: rankStr = "J"; break;
			default: rankStr = rank.ToString(); break; // 2-10 uses default
		}
		return rankStr + suit;	
	}

	// Create a Card Object from a string card "10D" - for file gamemode
	public static Card Parse(string cardStr) {
		char suit = cardStr[cardStr.Length - 1];
		string rankPart = cardStr.Substring(0, cardStr.Length - 1);

		int rank;
		switch(rankPart) {
			case "A": rank = 14; break;
			case "K": rank = 13; break;
			case "Q": rank = 12; break;
			case "J": rank = 11; break;
			default: rank = int.Parse(rankPart); break;
		}
		return new Card(rank, suit);
	}
}
		
